File: Bmi_2/app/src/main/java/com/pjatk/bmi_2/QuestionFragment.java
package com.pjatk.bmi_2;

import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import com.pjatk.bmi_2.model.Question;


public class QuestionFragment extends Fragment {
    private Question _question;
    private TextView _questionText;
    private QuizActivity _quizActivity;
    private Button[] _answers = new Button[4];

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.activity_question, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState){
        _questionText = view.findViewById(R.id.questionText);
        _answers[0] = view.findViewById(R.id.answer1);
        _answers[1] = view.findViewById(R.id.answer2);
        _answers[2] = view.findViewById(R.id.answer3);
        _answers[3] = view.findViewById(R.id.answer4);

        _questionText.setText(_question.getQuestionText());
        for (int i = 0; i < 4; i++){
            _answers[i].setEnabled(true);
            _answers[i].setText(_question.getAnswers().get(i).getAnswerText());
            _answers[i].setTag(i);
            _answers[i].setOnClickListener(answerClicked);
        }
    }

    public void SetArguments(Question question, QuizActivity quizActivity){
        _question = question;
        _quizActivity = quizActivity;
    }

    private View.OnClickListener answerClicked = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            int id = (int) v.getTag();

            v.setEnabled(false);

            if(_question.getAnswers().get(id).getIsCorrect()){
                _quizActivity.OnQuestionCorrect();
            }else{
                _quizActivity.OnQuestionBad();
            }
        }
    };
}

File: BMICalculator/app/src/main/java/com/szymon/bmicalculator/QuestionFragment.kt
package com.szymon.bmicalculator

import android.support.v4.app.Fragment
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView

import com.szymon.bmicalculator.model.Question


class QuestionFragment : Fragment() {
    private var _question: Question? = null
    private var _questionText: TextView? = null
    private var _quizActivity: QuizActivity? = null
    private val _answers = arrayOfNulls<Button>(4)

    private val answerClicked = View.OnClickListener { v ->
        val id = v.tag as Int

        v.isEnabled = false

        if (_question?.getAnswers()?.get(id)?.getIsCorrect()!!) {
            _quizActivity!!.onQuestionCorrect()
        } else {
            _quizActivity!!.onQuestionBad()
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.activity_question_fragment, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        _questionText = view.findViewById(R.id.questionText)
        _answers[0] = view.findViewById(R.id.answer1)
        _answers[1] = view.findViewById(R.id.answer2)
        _answers[2] = view.findViewById(R.id.answer3)
        _answers[3] = view.findViewById(R.id.answer4)

        _questionText!!.setText(_question!!.getQuestionText())
        for (i in 0..3) {
            _answers[i]?.isEnabled = true
            _answers[i]?.text = _question!!.getAnswers()[i].getAnswerText()
            _answers[i]?.tag = i
            _answers[i]?.setOnClickListener(answerClicked)
        }
    }

    fun SetArguments(question: Question, quizActivity: QuizActivity) {
        _question = question
        _quizActivity = quizActivity
    }
}